R/process-dir.R

```diff
@@ -0,0 +1,260 @@
+#' @param output_dir (`character(1)`) Optional overwrite directory where the output files should be saved. The default is `NULL`.
+#' By default the output directory is the same as the input directory.
+#' @param recurse (`logical(1)`) If `TRUE`, the function will search for files recursively in the input directory. The default is `FALSE`.
+#' @param flatten_output (`logical(1)`) If `TRUE`, the output files will be saved in the output directory directly. The default is `FALSE`.
+#' @param format (`character(1)`) The format of the Luminex data. The default is `NULL` and the format will have to
+#' be determined automatically based on the file name. Available options are `xPONENT` and `INTELLIFLEX`.
+#' @param layout_filepath (`character(1)`) The path to the layout file. The default is `NULL` and the layout file will have to
+#' be determined automatically based on the file name.
+#' @param normalisation_types (`character()`) A vector of normalisation types to use. The default is `c("RAU", "nMFI")`.
+#' @param generate_reports (`logical(1)`) If `TRUE`, generate quality control reports for each file. The default is `FALSE`.
+#' @param return_plates (`logical(1)`) If `TRUE`, return a list of processed plates. The default is `FALSE`.
+#' @param dry_run (`logical(1)`) If `TRUE`, the function will not process any files
+#' but will print the information about the files that would be processed. The default is `FALSE`.
+#' @param verbose (`logical(1)`) Print additional information. The default is `TRUE`.
+#' @param ... Additional arguments to for the `process_file` function.
+#'
+#' @import fs
+#'
+#' @export
+process_dir <- function(
+    input_dir,
+    output_dir = NULL,
+    recurse = FALSE,
+    flatten_output = FALSE,
+    layout_filepath = NULL,
+    format = NULL,
+    normalisation_types = c("RAU", "nMFI"),
+    generate_reports = FALSE,
+    return_plates = FALSE,
+    dry_run = FALSE,
+    verbose = TRUE,
+    ...) {
+  stopifnot(fs::dir_exists(input_dir))
+  stopifnot(is.null(output_dir) || fs::dir_exists(output_dir))
+  stopifnot(is.null(layout_filepath) || fs::file_exists(layout_filepath))
+  stopifnot(is.null(format) || (format %in% c("xPONENT", "INTELLIFLEX")))
+
+  input_files <- c()
+  for (input_file in fs::dir_ls(input_dir, recurse = recurse)) {
+    if (is_mbr_data_file(input_file)) {
+      input_files <- c(input_files, input_file)
+    }
+  }
+
+  if (length(input_files) == 0) {
+    verbose_cat("No files found in the input directory.\n", verbose = verbose)
+    return(NULL)
+  }
+
+  formats <- rep(NA, length(input_files))
+  for (i in seq_along(input_files)) {
+    formats[i] <- detect_mbr_format(input_files[i], format = format)
+  }
+  stopifnot(all(!is.na(formats)))
+
+  layouts <- rep(NA, length(input_files))
+  for (i in seq_along(input_files)) {
+    layouts[i] <- find_layout_file(
+      input_files[i],
+      layout_filepath = layout_filepath
+    )
+  }
+  stopifnot(all(!is.na(layouts)))
+
+  if (dry_run) {
```

Comment:
very good idea to include dry run option 👍

Reply:
Thank you!